src/comestible_inv_pane.cpp

```diff
@@ -0,0 +1,672 @@
+    const comestible_inv_area *sq = same_as_dragged ? get_square(
+                                        comestible_inv_area_info::AIM_DRAGGED ) : cur_area;
+
+    bool car = cur_area->has_vehicle() && viewing_cargo &&
+               sq->info.id != comestible_inv_area_info::AIM_DRAGGED;
+    auto name = utf8_truncate( sq->get_name( viewing_cargo ), width );
+    auto desc = utf8_truncate( sq->desc[car], width );
+    width -= 2 + 1; // starts at offset 2, plus space between the header and the text
+    mvwprintz( w, point( 2, 1 ), c_green, name );
+    mvwprintz( w, point( 2, 2 ), c_light_blue, desc );
+    trim_and_print( w, point( 2, 3 ), width, c_cyan, cur_area->flags );
+
+    for( auto &i : additional_info ) {
+        print_colored_text( window, i.position, i.color, i.color, i.message );
+    }
+
+    const int max_page = ( items.size() + itemsPerPage - 1 ) / itemsPerPage;
+    if( max_page > 1 ) {
+        const int page = index / itemsPerPage;
+        mvwprintz( w, point( 2, 4 ), c_light_blue, _( "[<] page %d of %d [>]" ), page + 1, max_page );
+    }
+
+    wattron( w, c_cyan );
+    // draw a darker border around the inactive pane
+    draw_border( w, BORDER_COLOR );
+    mvwprintw( w, point( 3, 0 ), _( "< [s]ort: %s >" ),
+               comestible_inv_listitem::get_col_data( sortby ).get_sort_name() );
+
+    if( !title.empty() ) {
+        std::string title_string = string_format( "<< %s >>", title );
+        mvwprintz( w, point( ( getmaxx( w ) - title_string.size() ) / 2, 0 ), c_light_cyan, title_string );
+    }
+
+    const char *fprefix = _( "[F]ilter" );
+    const char *fsuffix = _( "[R]eset" );
+
+    if( !filter_edit ) {
+        if( !filter.empty() ) {
+            mvwprintw( w, point( 2, getmaxy( w ) - 1 ), "< %s: %s >", fprefix, filter );
+        } else {
+            mvwprintw( w, point( 2, getmaxy( w ) - 1 ), "< %s >", fprefix );
+        }
+    }
+
+    wattroff( w, c_white );
+
+    if( !filter_edit && !filter.empty() ) {
+        mvwprintz( w, point( 6 + std::strlen( fprefix ), getmaxy( w ) - 1 ), c_white,
+                   filter );
+        mvwprintz( w, point( getmaxx( w ) - std::strlen( fsuffix ) - 2, getmaxy( w ) - 1 ), c_white, "%s",
+                   fsuffix );
+    }
+    wrefresh( w );
+}
+
+int comestible_inventory_pane::print_header( comestible_inv_area *sel_area )
+{
+    int wwidth = getmaxx( window );
+    int ofs = wwidth - 25 - 2 - 14;
+    for( size_t i = 0; i < all_areas->size(); ++i ) {
+        std::string key = comestible_inv_area::get_info(
+                              static_cast<comestible_inv_area_info::aim_location>( i ) ).minimapname;
+
+        comestible_inv_area *iter_square = get_square( i );
+
+        const char *bracket = iter_square->has_vehicle() ? "<>" : "[]";
+
+        bool is_single = cur_area->info.type != comestible_inv_area_info::AREA_TYPE_MULTI;
+        bool is_selected_by_user = cur_area->info.has_area( get_square( i )->info.id );
```

Comment:
```suggestion
        bool is_selected_by_user = cur_area->info.has_area( iter_square->info.id );
```
Avoids a repeated call to `get_square`.